acropolis-service/src/main/kotlin/org/ephyra/acropolis/service/impl/ExportService.kt

```diff
@@ -0,0 +1,127 @@
+package org.ephyra.acropolis.service.impl
+
+import org.ephyra.acropolis.external.*
+import org.ephyra.acropolis.external.model.ApplicationSoftware
+import org.ephyra.acropolis.external.model.Project
+import org.ephyra.acropolis.external.model.SoftwareContainer
+import org.ephyra.acropolis.external.model.SystemSoftware
+import org.ephyra.acropolis.persistence.api.ConnectionType
+import org.ephyra.acropolis.persistence.api.IConnectable
+import org.ephyra.acropolis.persistence.api.entity.ApplicationSoftwareEntity
+import org.ephyra.acropolis.persistence.api.entity.DatastoreEntity
+import org.ephyra.acropolis.persistence.api.entity.LoadBalancerEntity
+import org.ephyra.acropolis.persistence.api.entity.ProjectEntity
+import org.ephyra.acropolis.persistence.api.entity.QueueEntity
+import org.ephyra.acropolis.persistence.api.entity.ReverseProxyEntity
+import org.ephyra.acropolis.persistence.api.entity.SystemSoftwareEntity
+import org.ephyra.acropolis.service.api.IApplicationSoftwareService
+import org.ephyra.acropolis.service.api.IConnectionService
+import org.ephyra.acropolis.service.api.IExportService
+import org.ephyra.acropolis.service.api.IProjectService
+import org.ephyra.acropolis.service.api.ISystemSoftwareService
+import org.ephyra.acropolis.service.api.ImportType
+import org.springframework.stereotype.Service
+import java.lang.IllegalStateException
+import java.util.stream.Collectors
+
+/**
+ * Service for exporting project data to the Acropolis external model
+ */
+@Service
+class ExportService(
+        private val projectService: IProjectService,
+
+        private val applicationSoftwareService: IApplicationSoftwareService,
+
+        private val systemSoftwareService: ISystemSoftwareService,
+
+        private val connectionService: IConnectionService
+) : IExportService {
+    private val yamlHelper = YamlHelper()
+
+    override fun export(projectName: String, exportType: ImportType): String? {
+        val project = projectService.find(projectName)
+                ?: throw IllegalStateException("No project with name [$projectName]")
+
+        return when (exportType) {
+            ImportType.YAML -> {
+                yamlHelper.serialize(exportProject(project))
+            }
+            else -> {
+                throw IllegalStateException("Request to output to format [$exportType] which is not supported")
+            }
+        }
+    }
+
+    private fun exportProject(project: ProjectEntity): Project {
```

Comment:
Doc comments throughout.

Reply:
Generally don't want to put doc comments on private methods. It's ignored by documentation tools so you would only do it informally for other developers to read while working on the project. If your private methods need doc comments then they're probably too complicated though!